### PR TITLE
Allow setting of `copy_settings` in the HLRC

### PR DESCRIPTION
**Pull request by dliappis**
#30255 introduced the `copy_settings` parameter in 6.x.
This commit adds support for setting it via `setCopySettings` in the
HLRC.

### CODE REVIEW

**Comment by dliappis** [comment]
AFAIU (e.g. in https://github.com/elastic/elasticsearch/blob/73d474ef9c3f734dd52b1a6dedc72f04700a622b/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestResizeHandler.java#L60-L72) we don't need to backport this to 7.x/7.0 as we don't emit special deprecation warnings when it is unset.

Will be happy to add PRs if this is incorrect though.

**Comment by dliappis** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Comment by hub-cap** [comment]
@dliappis can you add some docs [here](https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java) as well pls?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java [hunk 1]
```diff
@@ -909,6 +909,17 @@ Params withWaitForActiveShards(ActiveShardCount activeShardCount) {
             return withWaitForActiveShards(activeShardCount, ActiveShardCount.DEFAULT);
         }
 
+        /**
+         * @deprecated <code>copy_settings</code> defaults to <code>true</code> and can not be set to false
```
Comment (jasontedor) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: Note that the default is unset and behaves as `false`, however `false` can not be set explicitly. `copy_settings` is either unset (default) or set to `true` (future behavior).

File: client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java [hunk 2]
```diff
@@ -868,6 +868,12 @@ private void resizeTest(ResizeType resizeType, CheckedFunction<ResizeRequest, Re
         ResizeRequest resizeRequest = new ResizeRequest(indices[0], indices[1]);
         resizeRequest.setResizeType(resizeType);
         Map<String, String> expectedParams = new HashMap<>();
+
+        if (ESTestCase.randomBoolean()) {
```
Comment (jasontedor) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java]: I see this is in the existing style in this class, but it is at odds with the rest of the codebase where we would simply write `randomBoolean()`.

**Review by jasontedor — COMMENTED** [review]
The production code looks good. I left a comment about the deprecation Javadoc, and I agree with @hub-cap to update the documentation that he referenced too.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java (hunk 1 above)
Comment (dliappis) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: Thanks great catch this got forgotten while I was looking the behavior of other branches.
Addressed in https://github.com/elastic/elasticsearch/pull/39752/commits/ce68119b9315a4dc812146e9e40b29cb8f9344f1

File: client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java (hunk 2 above)
Comment (dliappis) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java]: Addressed in https://github.com/elastic/elasticsearch/pull/39752/commits/ce68119b9315a4dc812146e9e40b29cb8f9344f1; in a separate PR we can convert everything in this class to `randomBoolean()` for consistency.

**Comment by dliappis** [comment]
Docs have been added in 5f05978

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java [hunk 3]
```diff
@@ -1683,9 +1683,13 @@ public void testShrinkIndex() throws Exception {
         request.setWaitForActiveShards(2); // <1>
         request.setWaitForActiveShards(ActiveShardCount.DEFAULT); // <2>
         // end::shrink-index-request-waitForActiveShards
+        // tag::shrink-index-request-copySettings
+        request.setCopySettings(Boolean.TRUE); // <1>
```
Comment (jasontedor) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java]: `true` is more natural

File: docs/java-rest/high-level/indices/shrink_index.asciidoc [hunk 4]
```diff
@@ -47,12 +47,21 @@ returns a response, as an `int`
 <2> The number of active shard copies to wait for before the shrink index API
 returns a response, as an `ActiveShardCount`
 
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request-copySettings]
+--------------------------------------------------
+<1> Use `Boolean.TRUE` to copy the settings from the source index to the target
```
Comment (jasontedor) [inline docs/java-rest/high-level/indices/shrink_index.asciidoc]: Use `true`

File: docs/java-rest/high-level/indices/shrink_index.asciidoc [hunk 5]
```diff
@@ -47,12 +47,21 @@ returns a response, as an `int`
...
+index. This cannot be `Boolean.FALSE`. If this method is not used, default behavior is not to copy.
```
Comment (jasontedor) [inline docs/java-rest/high-level/indices/shrink_index.asciidoc]: Maybe `false`?

**Review by jasontedor — COMMENTED** [review]
Looking good now, I left some minor feedback.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java (hunk 3 above)
Comment (dliappis) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java]: Addressed in de8ca1863c29ea222fe23b3366d207a4bdbcb4de

File: docs/java-rest/high-level/indices/shrink_index.asciidoc (hunk 4 above)
Comment (dliappis) [reply docs/java-rest/high-level/indices/shrink_index.asciidoc]: Addressed in de8ca1863c29ea222fe23b3366d207a4bdbcb4de

File: docs/java-rest/high-level/indices/shrink_index.asciidoc (hunk 5 above)
Comment (dliappis) [reply docs/java-rest/high-level/indices/shrink_index.asciidoc]: Addressed in de8ca1863c29ea222fe23b3366d207a4bdbcb4de

File: client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java [hunk 6]
```diff
@@ -868,6 +868,12 @@ private void resizeTest(ResizeType resizeType, CheckedFunction<ResizeRequest, Re
         ResizeRequest resizeRequest = new ResizeRequest(indices[0], indices[1]);
         resizeRequest.setResizeType(resizeType);
         Map<String, String> expectedParams = new HashMap<>();
+
+        if (randomBoolean()) {
+            resizeRequest.setCopySettings(Boolean.TRUE);
```
Comment (jasontedor) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java]:
```suggestion
            resizeRequest.setCopySettings(true);
```

File: docs/java-rest/high-level/indices/shrink_index.asciidoc [hunk 7]
```diff
@@ -47,12 +47,21 @@ returns a response, as an `int`
 <2> The number of active shard copies to wait for before the shrink index API
 returns a response, as an `ActiveShardCount`
 
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request-copySettings]
+--------------------------------------------------
+<1> Use `true` to copy the settings from the source index to the target
+index. This cannot be `false`. If this method is not used, default behavior is not to copy.
```
Comment (jasontedor) [inline docs/java-rest/high-level/indices/shrink_index.asciidoc]: Can you wrap to 80 columns?

**Review by jasontedor — APPROVED** [review]
I left two more minors, no need for another review round. LGTM. Fire at will.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java (hunk 6 above)
Comment (dliappis) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesRequestConvertersTests.java]: Fixed in https://github.com/elastic/elasticsearch/pull/39752/commits/b20c245d526abdad6902a2d467b1e1ef9c30ab76

File: docs/java-rest/high-level/indices/shrink_index.asciidoc (hunk 7 above)
Comment (dliappis) [reply docs/java-rest/high-level/indices/shrink_index.asciidoc]: Fixed in https://github.com/elastic/elasticsearch/pull/39752/commits/b20c245d526abdad6902a2d467b1e1ef9c30ab76

**Comment by dliappis** [comment]
> I left two more minors, no need for another review round. LGTM. Fire at will.

Thanks a lot for the review effort here. Will merge after CI is green.

**Pull request merged**